docs/docs/composition-vs-inheritance.md

```diff
@@ -151,7 +151,7 @@ class SignUpDialog extends React.Component {
   }
 
   handleChange(e) {
-    this.setState({login: e.target.value});
+    this.setState(() => ({login: e.target.value}));
```

Comment:
I'm not sure if this will work. The synthetic event may be unavailable at this point and be recycled if in the pool. In fact, you don't need to use functional setState at all in the case. Not using it in this case is completely fine.